lib/ember-cli/deploy/redis.rb

```diff
@@ -0,0 +1,82 @@
+require "active_support/core_ext/object/blank"
+require "ember-cli/deploy/page"
+require "redis"
+
+module EmberCLI
+  module Deploy
+    class Redis
+      def initialize(namespace:, index_html: nil)
```

Comment:
@seanpdoyle This is the first time I've done a connection pool, so I'm sure there's some areas for improvement. I used the [ `connection_pool`](https://github.com/mperham/connection_pool) gem from @mperham.

lib/redis_connection_pool.rb

```ruby
class RedisConnectionPool
  extend SingleForwardable

  mattr_accessor :size
  self.size = 1

  def_delegator :pool, :with

  private

  def self.pool
    @pool ||= ConnectionPool.new(size: size)  { Redis.new url: ENV['REDIS_URL'] }
  end
end
```

config/puma.rb

```ruby
workers Integer(ENV['WEB_CONCURRENCY'] || 2)
threads_count = Integer(ENV['MAX_THREADS'] || 5)
threads threads_count, threads_count

preload_app!

rackup      DefaultRackup
port        ENV['PORT']     || 3000
environment ENV['RACK_ENV'] || 'development'

on_worker_boot do
  # Worker specific setup for Rails 4.1+
  # See: https://devcenter.heroku.com/articles/deploying-rails-applications-with-the-puma-web-server#on-worker-boot
  ActiveRecord::Base.establish_connection

  RedisConnectionPool.size = threads_count
end
```

app/controllers/ember_controller.rb

```ruby
require 'ember-cli/deploy/redis'

class EmberController < ApplicationController

  def frontend
    RedisConnectionPool.with do |redis|
      @deploy = EmberCLI::Deploy::Redis.new(
        redis_client: redis
      )

      render text: @deploy.html, layout: false
    end
  end
end
```

This fixed my issue where I was running out of connections, and it's running in production now. If you have some suggestions for improvement, I'd be happy to hear them.

Also, if you want to merge this pull request, I could start working on a new pull request to document strategies for managing the html from rails and the html from ember.